server/apps/publish/formatters/newsml_g2_formatter.py

```diff
@@ -34,21 +35,35 @@ class NewsMLG2Formatter(Formatter):
     def format(self, article, subscriber):
         try:
             pub_seq_num = superdesk.get_resource_service('subscribers').generate_sequence_number(subscriber)
-
-            nitfFormater = NITFFormatter()
-            nitf = nitfFormater.get_nitf(article, subscriber, pub_seq_num)
-
+            is_package = self._is_package(article)
             self._message_attrib.update(self._debug_message_extra)
             newsMessage = etree.Element('newsMessage', attrib=self._message_attrib)
             self._format_header(article, newsMessage, pub_seq_num)
             itemSet = self._format_item(newsMessage)
-            if article[ITEM_TYPE] in [CONTENT_TYPE.TEXT, CONTENT_TYPE.PREFORMATTED]:
-                self._format_newsitem(article, itemSet, nitf)
+            if is_package:
+                item = self._format_item_set(article, itemSet, 'packageItem')
+                self._format_groupset(article, item)
+            elif article[ITEM_TYPE] == CONTENT_TYPE.PICTURE:
+                item = self._format_item_set(article, itemSet, 'newsItem')
+                self._format_contentset(article, item)
+            else:
+                nitfFormater = NITFFormatter()
+                nitf = nitfFormater.get_nitf(article, subscriber, pub_seq_num)
+                newsItem = self._format_item_set(article, itemSet, 'newsItem')
+                self._format_content(article, newsItem, nitf)
 
             return [(pub_seq_num, self.XML_ROOT + etree.tostring(newsMessage).decode('utf-8'))]
         except Exception as ex:
             raise FormatterError.newmsmlG2FormatterError(ex, subscriber)
 
+    def _is_package(self, article):
+        """
+        Given an article returns if it is a none takes package or not
+        :param artcile:
+        :return: True is package
+        """
+        return article[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE and article.get(PACKAGE_TYPE, '') == ''
```

Comment:
@marwoodandrew is this function only for the Takes Package a.k.a digital master file.